### PR TITLE
[SPARK-39111][CORE][SQL] Mark overridden methods with `@Override` annotation

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

This PR marks overridden methods with `@Override` annotation. 

### Why are the changes needed?

An overridden method from an interface or abstract class must be marked with `@Override` annotation. To accurately determine whether the overriding is successful, an `@Override` annotation is necessary. Meanwhile, once the method signature in the abstract class is changed, the implementation class will report a compile-time error immediately.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the unit test.

### CODE REVIEW

**Review by srowen — COMMENTED** [review]
That's fine. Are there any other missing Override annotations? if you're already running a static analyzer

**Comment by dcoliversun** [comment]
@srowen Hi, I check it again and mark all overridden methods with `@override` annotation

File: common/network-common/src/main/java/org/apache/spark/network/util/NettyLogger.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ private static class NoContentLoggingHandler extends LoggingHandler {
       super(clazz, level);
     }
 
+    @Override
```
Comment (bjornjorgensen) [inline common/network-common/src/main/java/org/apache/spark/network/util/NettyLogger.java:38]: OK found the same with sonarqube

File: connector/avro/src/main/java/org/apache/spark/sql/avro/SparkAvroKeyOutputFormat.java [hunk 2]
```diff
@@ -46,6 +46,7 @@ static class SparkRecordWriterFactory extends RecordWriterFactory<GenericRecord>
       this.metadata = metadata;
     }
 
+    @Override
```
Comment (bjornjorgensen) [inline connector/avro/src/main/java/org/apache/spark/sql/avro/SparkAvroKeyOutputFormat.java:49]: OK found the same with sonarqube

File: launcher/src/main/java/org/apache/spark/launcher/AbstractLauncher.java [hunk 3]
```diff
@@ -298,6 +298,7 @@ protected boolean handleUnknown(String opt) {
       return true;
     }
 
+    @Override
```
Comment (bjornjorgensen) [inline launcher/src/main/java/org/apache/spark/launcher/AbstractLauncher.java:301]: OK found the same with sonarqube

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/AlwaysFalse.java [hunk 4]
```diff
@@ -34,13 +34,16 @@ public AlwaysFalse() {
...
   public Boolean value() {
     return false;
   }
 
+  @Override
   public DataType dataType() {
     return DataTypes.BooleanType;
   }
 
+  @Override
```
Comment (bjornjorgensen) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/AlwaysFalse.java:47]: OK found the same with sonarqube

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/AlwaysTrue.java [hunk 5]
```diff
@@ -34,13 +34,16 @@ public AlwaysTrue() {
...
   public Boolean value() {
     return true;
   }
 
+  @Override
   public DataType dataType() {
     return DataTypes.BooleanType;
   }
 
+  @Override
```
Comment (bjornjorgensen) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/AlwaysTrue.java:47]: OK found the same with sonarqube

**Review by bjornjorgensen — COMMENTED** [review]
This it what sonarqube have marked as "Using the @Override annotation is useful for two reasons :

It elicits a warning from the compiler if the annotated method doesn’t actually override anything, as in the case of a misspelling.
It improves the readability of the source code by making it obvious that methods are overridden."

**Review by srowen — APPROVED** [review]
Looks good, it is a good practice to use this annotation

**Comment by srowen** [comment]
Merged to master

**Comment by dcoliversun** [comment]
Thank you @srowen @bjornjorgensen

**Pull request closed without merging**